dsperse/src/prove/prover.py

```
@@ -0,0 +1,314 @@
+    def _execute_proving(self, work_items: list) -> list:
+        """Execute proving tasks in parallel or sequentially."""
+        if self.parallel > 1 and len(work_items) > 1:
+            logger.info(f"Proving {len(work_items)} slices with {self.parallel} parallel processes...")
+            results = []
+            with ProcessPoolExecutor(max_workers=self.parallel) as executor:
+                futures = {executor.submit(ProverUtils.prove_slice_logic, item): item[0] for item in work_items}
+                for future in as_completed(futures):
+                    slice_id = futures[future]
+                    try:
+                        results.append(future.result())
+                    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)